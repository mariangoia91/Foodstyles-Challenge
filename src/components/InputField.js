import React from "react"
import { StyleSheet, View, Text, TextInput } from "react-native"

const InputField = props => {
  return (
    <View
      style={[
        { width: "100%", justifyContent: "center", alignItems: "center" }
      ]}
    >
      <Text style={[styles.fieldText, { marginBottom: 8 }]}>{props.title}</Text>
      <TextInput
        style={[styles.input, styles.roundedCorners, { marginBottom: 24 }]}
        keyboardType={props.keyboardType}
        secureTextEntry={props.isSecure}
        autoCorrect={false}
        autoCapitalize={props.autoCapitalize}
        onChangeText={text => props.mutator(text)}
        value={props.value}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  fieldText: {
    fontFamily: "Montserrat-Regular",
    width: "80%",
    textAlign: "left",
    fontSize: 18,
    color: "white"
  },
  roundedCorners: {
    borderRadius: 8
  },
  input: {
    fontFamily: "Montserrat-Regular",
    width: "80%",
    height: 45,
    backgroundColor: "white",
    fontWeight: "600",
    paddingHorizontal: 8
  }
})

export default InputField
