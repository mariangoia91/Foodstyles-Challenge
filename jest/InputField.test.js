import React from "react"
import InputField from "../src/components/InputField"

import renderer from "react-test-renderer"

test("renders correctly", () => {
  const tree = renderer.create(<InputField />).toJSON()
  expect(tree).toMatchSnapshot()
})
