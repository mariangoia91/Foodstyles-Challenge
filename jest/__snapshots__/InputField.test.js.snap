// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Montserrat-Regular",
          "fontSize": 18,
          "textAlign": "left",
          "width": "80%",
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "fontFamily": "Montserrat-Regular",
          "fontWeight": "600",
          "height": 45,
          "paddingHorizontal": 8,
          "width": "80%",
        },
        Object {
          "borderRadius": 8,
        },
        Object {
          "marginBottom": 24,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
